### DISCUSSION TITLE
Is there a way to make paginate middleware example work?

### DISCUSSION THREAD

**Discussion by msurmeli** [go-chi/chi#751] [Q&A]
Considering that response writer and request are the only objects that we can take or pass data, how this example supposed to work?

https://github.com/go-chi/chi/blob/master/_examples/rest/main.go#L252
The function above, to be precise.

**Comment by jhhom** [comment]
I think one way I would do this is I would get the pagination information (e.g page number, page size) from the URL query or request payload (probably decode them if needed), then store in the request context so that the next handler can access these information.

Some examples:
1. [How to store information in the request context](https://github.com/go-chi/chi/blob/master/_examples/rest/main.go#L146)
2. [How to access that information from the handler](https://github.com/go-chi/chi/blob/master/_examples/rest/main.go#L181)
